### ISSUE TITLE
Enhance autocorrect prompt to add to CORRECT_IGNORE or CORRECT_IGNORE_FILE variable

### ISSUE THREAD

**Issue by BrianCanFixIT** [ohmyzsh/ohmyzsh#9571]
**Issue**
Autocorrect is awesome, but often users of zsh will end up turning it off because they get a correction that annoys them and they do not have an easy way to tell the system to ignore that correction.
```
user@computer ~$ killall ssh
zsh: correct 'ssh' to '.ssh' [nyae]? n
```
One such example:
https://stackoverflow.com/questions/3437454/zsh-wants-to-autocorrect-a-command-with-an-before-it

**Solution**
When prompted for an autocorrect, have an option to "**I**gnore" this particular correction.
```
user@computer ~$ killall ssh
zsh: correct 'ssh' to '.ssh' [nyaei]? n
```
Choosing the "i" option will add the particular correction to the CORRECT_IGNORE or CORRECT_IGNORE_FILE variable in the ~/.zshrc file.

In the particular example above the following would be set in the ~/.zshrc file:
```
CORRECT_IGNORE_FILE=".ssh"
```

If there was a value for CORRECT_IGNORE_FILE already, the code would attempt a simple regex pipe followed by the pattern to be added to the ignore list in the ~/.zshrc file:
```
CORRECT_IGNORE_FILE=".somefile|.ssh"
```

**Comment by mcornella** [comment]
If you have `ENABLE_CORRECTION=true` enabled, you can unset it and use `setopt correct` instead, which will only correct command names and not parameters. What you're asking for would be a zsh feature.

**Comment by BrianCanFixIT** [comment]
The point is to NOT disable corrections for parameters. 
The point is to enhance the user experience so that the correction function is enhanced rather than disabled for parameters and commands alike.

**Comment by mcornella** [comment]
Right. As I said though, this would be zsh territory. I don't have the bandwidth to code such a feature, [ask in the mailing list](zsh.org/mla).

**Comment by CantoroMC** [comment]
alias ssh='nocorrect ssh'

**Comment by justrajdeep** [comment]
i tried 

    alias ...='nocorrect ...'
    
but when i tried `p4 opened ...` 

it still tried to correct ... am i missing something?

**Comment by robbyrussell** [comment]
Thanks for writing this up, and sorry it sat here so long without a proper answer.

Marc's reply back in 2021 is still the right one... that `[nyae]` prompt is built into zsh itself, not something Oh My Zsh puts on screen. All we do is turn on `correct_all` and ship a handful of `nocorrect` aliases in `lib/correction.zsh`. An extra key that appends the word to `CORRECT_IGNORE` would need a change to zsh's own source, so that conversation has to happen on the zsh-workers list rather than here.

Closing it out so it's not sitting in limbo. It's a reasonable idea, just not one we're able to build.

🤖 Co-authored with Claude Code
